scripts/test-pipelines.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+shopt -s globstar
+shopt -s nullglob
+
+# Quote the heredoc to prevent shell expansion
+cat << "EOF"
+workflow:
+  rules:
+    # Disable merge request pipelines
+    - if: $CI_MERGE_REQUEST_ID
+      when: never
+    - when: always
+default:
+  interruptible: true
+variables:
+  GH_PROJECT_PATH: "MatrixAI/${CI_PROJECT_NAME}"
+  GH_PROJECT_URL: "https://${GITHUB_TOKEN}@github.com/${GH_PROJECT_PATH}.git"
+  GIT_SUBMODULE_STRATEGY: "recursive"
+  # Cache .npm
+  NPM_CONFIG_CACHE: "./tmp/npm"
+  # Prefer offline node module installation
+  NPM_CONFIG_PREFER_OFFLINE: "true"
+  # `ts-node` has its own cache
+  # It must use an absolute path, otherwise ts-node calls will CWD
+  TS_CACHED_TRANSPILE_CACHE: "${CI_PROJECT_DIR}/tmp/ts-node-cache"
+  TS_CACHED_TRANSPILE_PORTABLE: "true"
+# Cached directories shared between jobs & pipelines per-branch per-runner
+cache:
+  key: $CI_COMMIT_REF_SLUG
+  paths:
+    - ./tmp/npm/
+    - ./tmp/ts-node-cache/
+    # `jest` cache is configured in jest.config.js
+    - ./tmp/jest/
+stages:
+  - check       # Linting, unit tests
+image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
+EOF
+
+printf "\n"
+
+# Using shards to optimise tests
+# In the future we can incorporate test durations rather than using
+# a static value for the parallel keyword
+cat << "EOF"
+check:test:
+  stage: check
+  needs: []
+  parallel: 2
+  script:
+    - >
+        nix-shell --run '
+        npm run build --verbose;
```

Comment:
I just realised that during testing, you may not actually need this. But I had this enabled because in TS demo lib native, you do need to build, before running tests as it builds the native addon too. But in ts demo lib, there's no native addon to build.